server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java

```diff
@@ -775,4 +786,20 @@ public static boolean isRealmProviderJpa(KeycloakSession session) {
         return providerIds != null && providerIds.size() == 1 && providerIds.iterator().next().equals("jpa");
     }
 
+    /**
+     * @param clientAuthenticatorType
+     * @return secret size based on authentication type
+     */
+    public static int getSecretLengthByAuthenticationType(String clientAuthenticatorType, String signingAlg) {
+        String sanitizedClientAuthType = clientAuthenticatorType.replaceAll("[^a-zA-Z0-9]",""); //FIXME some locations are referenced with "_" while others with "-"
```

Comment:
This is a bit strange. I see that `JWTClientSecretAuthenticator.PROVIDER_ID` is `client-secret-jwt` and that is how is the factory selected in various places in the code (For example in `ClientManager.getClientCredentialsAdapterConfig` method).

I hope this workaround is not needed. If there are any places in the code or any failing tests without this workaround, it rather worth to investigate why the `_` is needed instead of `-` .

Reply:
Workaround removed. Let's just treat it as client-secret-jwt. Fixed bug in base tests